### PR TITLE
CIV-11832 Enabling View Documents Task List After Evidence Upload

### PR DESCRIPTION
**Pull request by kannan-v-hmcts**
PR for https://tools.hmcts.net/jira/browse/CIV-11832

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/dashboard/scenarios/claimant/EvidenceUploadedClaimantScenarioTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        handler.handle(callbackParams(caseData));
+
+        //Verify Task List
+
+        doGet(BEARER_TOKEN, GET_TASKS_ITEMS_URL, caseId, "CLAIMANT")
+            .andExpectAll(
+                status().is(HttpStatus.OK.value()),
+                jsonPath("$[0].reference").value(caseId.toString()),
+                jsonPath("$[0].taskNameEn").value("<a href=\"{VIEW_EVIDENCE_UPLOAD_DOCUMENTS}\" class=\"govuk-link\">View documents</a>"),
+                jsonPath("$[0].currentStatusEn").value("Available")
```
Comment (miguelMolina3691) [inline src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/dashboard/scenarios/claimant/EvidenceUploadedClaimantScenarioTest.java:81]: Could you add welsh test?

File: src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/dashboard/scenarios/defendant/EvidenceUploadedDefendantScenarioTest.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        handler.handle(callbackParams(caseData));
+
+        //Verify Task List
+
+        doGet(BEARER_TOKEN, GET_TASKS_ITEMS_URL, caseId, "DEFENDANT")
+            .andExpectAll(
+                status().is(HttpStatus.OK.value()),
+                jsonPath("$[0].reference").value(caseId.toString()),
+                jsonPath("$[0].taskNameEn").value("<a href=\"{VIEW_EVIDENCE_UPLOAD_DOCUMENTS}\" class=\"govuk-link\">View documents</a>"),
+                jsonPath("$[0].currentStatusEn").value("Available")
```
Comment (miguelMolina3691) [inline src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/dashboard/scenarios/defendant/EvidenceUploadedDefendantScenarioTest.java:81]: Could you add welsh test?

File: src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/dashboard/scenarios/claimant/EvidenceUploadedClaimantScenarioTest.java (hunk 1 above)
Comment (kannan-v-hmcts) [reply src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/dashboard/scenarios/claimant/EvidenceUploadedClaimantScenarioTest.java:81]: Good catch. I missed it. Will add 👍🏻

**Review by laarmada — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/defendant/EvidenceUploadedDefendantNotificationHandler.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        return EVENTS;
+    }
+
+    @Override
+    protected String getScenario(CaseData caseData) {
+        if (nonNull(caseData.getCaseDocumentUploadDateRes())) {
+            return DashboardScenarios.SCENARIO_AAA6_CP_HEARING_DOCUMENTS_DEFENDANT_UPLOADED_DEFENDANT.getScenario();
+        } else {
+            return DashboardScenarios.SCENARIO_AAA6_CP_HEARING_DOCUMENTS_CLAIMANT_UPLOADED_DEFENDANT.getScenario();
+        }
```
Comment (m-meulendijks-v1) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/defendant/EvidenceUploadedDefendantNotificationHandler.java:46]: Would it be more readable to do `return nonNull() ? scenario 1 : scenario 2;` ?
Not a necessary change, but a suggestion.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/claimant/EvidenceUploadedClaimantNotificationHandler.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        return EVENTS;
+    }
+
+    @Override
+    protected String getScenario(CaseData caseData) {
+        if (nonNull(caseData.getCaseDocumentUploadDate())) {
+            return DashboardScenarios.SCENARIO_AAA6_CP_HEARING_DOCUMENTS_CLAIMANT_UPLOADED_CLAIMANT.getScenario();
+        } else {
+            return DashboardScenarios.SCENARIO_AAA6_CP_HEARING_DOCUMENTS_DEFENDANT_UPLOADED_CLAIMANT.getScenario();
+        }
```
Comment (m-meulendijks-v1) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/claimant/EvidenceUploadedClaimantNotificationHandler.java:46]: Would it be more readable to do `return nonNull() ? scenario 1 : scenario 2;` ?
Not a necessary change, but a suggestion.

File: src/main/resources/db/migration/V2024_05_15_1300__CIV-11832_hearing_documents_uploaded_scenario.sql [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+       ('Scenario.AAA6.CP.HearingDocuments.Claimant.Uploaded.Defendant',
+        '{}',
+        '{}'),
+       ('Scenario.AAA6.CP.HearingDocuments.Defendant.Uploaded.Claimant',
+        '{}',
+        '{}'),
+       ('Scenario.AAA6.CP.HearingDocuments.Defendant.Uploaded.Defendant',
+        '{Notice.AAA6.CP.HearingDocuments.Upload.Defendant}',
+        '{}');
+
```
Comment (m-meulendijks-v1) [inline src/main/resources/db/migration/V2024_05_15_1300__CIV-11832_hearing_documents_uploaded_scenario.sql:17]: I wonder if the naming is confusing - although I have no issues with the functionality, and will still approve.

Should we have it more like:
1. Scenario.AAA6.CP.HearingDocuments.Uploaded.Claimant -> When claimant has uploaded evidence at some point.
2. Scenario.AAA6.CP.HearingDocuments.NotUploaded.Claimant -> When claimant has uploaded nothing yet, but their dashboard needs updating.
3. Scenario.AAA6.CP.HearingDocuments.Uploaded.Defendant -> When defendant has uploaded evidence already, and update required.
4. Scenario.AAA6.CP.HearingDocuments.NotUploaded.Defendant -> When defendant has not uploaded yet, but dashboard update required.

I feel that change matches your actual implementation better (the current naming might be a holdover from the approach we initially discussed).

**Review by m-meulendijks-v1 — DISMISSED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/dashboard/scenarios/claimant/EvidenceUploadedClaimantScenarioTest.java (hunk 1 above)
Comment (kannan-v-hmcts) [reply src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/dashboard/scenarios/claimant/EvidenceUploadedClaimantScenarioTest.java]: Done

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/defendant/EvidenceUploadedDefendantNotificationHandler.java (hunk 3 above)
Comment (kannan-v-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/defendant/EvidenceUploadedDefendantNotificationHandler.java]: Done

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/claimant/EvidenceUploadedClaimantNotificationHandler.java (hunk 4 above)
Comment (kannan-v-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/claimant/EvidenceUploadedClaimantNotificationHandler.java]: Done

File: src/main/resources/db/migration/V2024_05_15_1300__CIV-11832_hearing_documents_uploaded_scenario.sql (hunk 5 above)
Comment (kannan-v-hmcts) [reply src/main/resources/db/migration/V2024_05_15_1300__CIV-11832_hearing_documents_uploaded_scenario.sql:17]: Your suggestion sounds more readable. Changes done

**Review by miguelMolina3691 — DISMISSED** [review]

**Review by m-meulendijks-v1 — DISMISSED** [review]

**Review by dharmendrak — APPROVED** [review]

**Review by dharmendrak — APPROVED** [review]

**Review by laarmada — APPROVED** [review]

**Review by m-meulendijks-v1 — APPROVED** [review]

**Pull request merged**